emailer.py

```diff
@@ -0,0 +1,167 @@
+    }
+
+    SUBJECTS = {
+        NOTIFICATION: DEFAULT_NOTIFICATION_SUBJECT,
+        VALIDATION : DEFAULT_VALIDATION_SUBJECT,
+    }
+
+    @classmethod
+    def from_sitewide_integration(cls, _db):
+        """Create an Emailer from a site-wide email integration.
+
+        :param _db: A database connection
+        """
+        integration = cls._sitewide_integration(_db)
+        host = integration.url
+        port = integration.setting(cls.PORT).int_value or 587
+        from_address = integration.setting(cls.FROM_ADDRESS).value
+
+        email_templates = {}
+        for email_type in cls.EMAIL_TYPES:
+            subject = (
+                integration.setting(email_type + "_subject").value or
+                cls.SUBJECTS[email_type]
+            )
+            body = (
+                integration.setting(email_type + "_body").value or
+                cls.BODIES[email_type]
+            )
+            email_templates[email_type] = EmailTemplate(subject, body)
+
+        return cls(smtp_username=integration.username,
+                   smtp_password=integration.password,
+                   smtp_host=host, smtp_port=port, from_address=from_address,
+                   templates=email_templates)
+
+    @classmethod
+    def _sitewide_integration(cls, _db):
+        """Find the ExternalIntegration for the emailer."""
+        from model import ExternalIntegration
+        qu = _db.query(ExternalIntegration).filter(
+            ExternalIntegration.goal==cls.GOAL
+        )
+        integrations = qu.all()
+        if not integrations:
+            raise CannotLoadConfiguration(
+                "No email integration is configured."
+            )
+            return None
+
+        if len(integrations) > 1:
+            # If there are multiple integrations configured, none of
+            # them can be the 'site-wide' configuration.
+            raise CannotLoadConfiguration(
+                'Multiple email integrations are configured'
+            )
+
+        [integration] = integrations
+        return integration
+
+    def __init__(self, smtp_username, smtp_password, smtp_host, smtp_port,
+                 from_address, templates):
+        """Constructor."""
+        if not smtp_username:
+            raise CannotLoadConfiguration("No SMTP username specified")
+        self.smtp_username = smtp_username
+        if not smtp_password:
+            raise CannotLoadConfiguration("No SMTP password specified")
+        self.smtp_password = smtp_password
+        if not smtp_host:
+            raise CannotLoadConfiguration("No SMTP host specified")
+        self.smtp_host = smtp_host
+        if not smtp_host:
+            raise CannotLoadConfiguration("No SMTP port specified")
+        self.smtp_port = smtp_port
+        if not from_address:
+            raise CannotLoadConfiguration("No From: address specified")
+        self.from_address = from_address
+        self.templates = templates
```

Comment:
It might be good to verify that no unexpected arguments are needed by the templates.